packages/cli/src/commands/common/accounts.ts

```diff
@@ -29,6 +30,31 @@ export const ACCOUNTS_OPTION: KeyedOption<AccountsArg> = {
   type: 'stringsList',
 }
 
+export const getAdaptersTags = (adapters: string[]): Tags => (
+  Object.fromEntries(adapters
+    .filter(adapter => getSupportedServiceAdapterNames().includes(adapter))
+    .map(adapter => [`adapter-${adapter}`, true]))
+)
+
+const getValidAccounts = (ws: Workspace, accounts?: string[], env?: string): string[] => {
+  const validAccounts = ws.accounts(env)
+  return _.difference(validAccounts, accounts || [])
+}
+
+export const getTagsForAccounts = (
+  ws: Workspace,
+  accounts?: string[],
+  env?: string
+): Tags => getAdaptersTags(getValidAccounts(ws, accounts, env).map(ws.getServiceFromAccountName))
+
+export const getTagsForInputAccounts = (
+  ws: Workspace,
+  input: {
+    accounts?: string[]
+    env?: string
+  }
```

Comment:
should we be using the types? I think we may want to couple this explicitly instead of repeating the "accounts" and "env" name...
```suggestion
  input: AccountsArg & EnvArg
```